search/query.go

```diff
@@ -114,7 +121,45 @@ func ParseQuery(q string) Query {
 	return Query{Term: strings.Join(terms, " "), ExactTerms: exactTerms, Original: q, Filters: filters}
 }
 
-func createResultsQuery(resultTypes []string, q Query, docIds []string, filterOutCUSources []string) elastic.Query {
+func createSpanNearQuery(field string, term string, boost float32, slop int, inOrder bool) elastic.Query {
+	clauses := make([]string, 0)
+	spanNearMask := "{\"span_near\": { \"clauses\": [%s], \"slop\": %d, \"boost\": %f, \"in_order\": %t } }"
+	clauseMask := "{\"span_multi\": { \"match\": { \"fuzzy\": { \"%s\": { \"value\": \"%s\", \"fuzziness\": %s, \"transpositions\": %s } } } } }"
+	splitted := strings.Fields(term)
+	for _, t := range splitted {
+		if t == "<" || t == ">" || t == "-" {
+			continue
+		}
+		fuzzines := "\"AUTO\""   // Default.
+		transpositions := "true" // Default.
+		runes := []rune(t)
+		_, convertToIntErr := strconv.Atoi(t)
+		if convertToIntErr == nil || (len(runes) == 3 && runes[1] == '"') || (len(runes) == 4 && runes[2] == '"') {
```

Comment:
This must be extracted and tested in a function because:
1) It is easy (unit test, no end to end test).
2) Critical and error prone.

Reply:
Add tests task